fe/fe-core/src/test/java/com/starrocks/clone/DynamicPartitionSchedulerTest.java

```diff
@@ -355,4 +495,251 @@ public void testDuplicatePartitionException() throws Exception {
             fail("Should not throw exception: " + e.getMessage());
         }
     }
+
+    @Test
+    public void testListPartitionTTLCondition1() {
+        for (String t : LIST_PARTITION_TABLES) {
+            starRocksAssert.withTable(t,
+                    (obj) -> {
+                        String tableName = (String) obj;
+                        withTableListPartitions(tableName);
+                        OlapTable olapTable = (OlapTable) starRocksAssert.getTable("test", tableName);
+                        Assert.assertEquals(4, olapTable.getVisiblePartitions().size());
+                        String dropPartitionSql = String.format("alter table %s set ('partition_retention_condition' = " +
+                                "'dt >= current_date() - interval 1 month');", tableName);
+                        starRocksAssert.alterTable(dropPartitionSql);
+
+                        DynamicPartitionScheduler scheduler = GlobalStateMgr.getCurrentState()
+                                .getDynamicPartitionScheduler();
+                        Database db = GlobalStateMgr.getCurrentState().getLocalMetastore().getDb("test");
+                        OlapTable tbl = (OlapTable) GlobalStateMgr.getCurrentState().getLocalMetastore()
+                                .getTable(db.getFullName(), tableName);
+                        try {
+                            scheduler.runOnceForTest();
+                            Assert.assertTrue(tbl.getVisiblePartitions().size() == 0);
+                            // add a new partition and an expired partition
+                            LocalDateTime now = LocalDateTime.now();
+                            addListPartition(tableName, "p5", "guangdong",
+                                    now.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
+                            addListPartition(tableName, "p6", "guangdong",
+                                    now.minusMonths(1).format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
+                            Assert.assertTrue(tbl.getVisiblePartitions().size() == 2);
+
+                            scheduler.runOnceForTest();
+                            Assert.assertTrue(tbl.getVisiblePartitions().size() == 2);
+                        } catch (Exception e) {
```

Comment:
is it necessary ? the test will fail if it throw exceptions, i don't think it's need to catch exceptions manually